### PR TITLE
Update constraints in order to allow some "symfony/*:^5.2" deps

### PR DESCRIPTION
**Pull request by phansys**
## Subject
Update constraints in order to allow some "symfony/*:^5.2" deps.

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Added
- Support for "symfony/form:^5.2".
```

## To do

- [ ] Check if we can allow more "symfony/*:^5.2" dependencies.

### CODE REVIEW

**Comment by dmaicher** [comment]
Wondering: Shouldn't we put our efforts into releasing 4.0 instead? It fully supports Symfony 5 already.

**Comment by phansys** [comment]
IMO, the easiest way to release 4.0 is being sure that the majority of dependencies it will support are already proven.

**Comment by VincentLanglet** [comment]
> IMO, the easiest way to release 4.0 is being sure that the majority of dependencies it will support are already proven.

We already have a lot of user with the master branch, which are using Symfony 5

File: composer.json [hunk 1]
```diff
@@ -37,31 +37,31 @@
...
         "symfony/console": "^4.4",
-        "symfony/dependency-injection": "^4.4.8 || ^5.1",
+        "symfony/dependency-injection": "^4.4.8 || ^5.2",
         "symfony/doctrine-bridge": "^4.4",
         "symfony/event-dispatcher": "^4.4",
         "symfony/event-dispatcher-contracts": "^1.1 || ^2.0",
-        "symfony/expression-language": "^4.4 || ^5.1",
-        "symfony/form": "^4.4.12",
+        "symfony/expression-language": "^4.4 || ^5.2",
+        "symfony/form": "^4.4.12 || ^5.2",
```
Comment (dmaicher) [inline composer.json:48]: I think this one is actually tricky with keeping BC. It was also discussed here: https://github.com/sonata-project/SonataAdminBundle/pull/6362 (was switched from 3.x to master)

Comment (phansys) [reply composer.json:48]: Sure, that's why this PR is kept as draft yet.

**Comment by VincentLanglet** [comment]
Can we close this @phansys ? I would say we're really close to Sonata 4.

**Pull request closed without merging**